### PR TITLE
[CMSIS-NN] Align CMSIS-NN in TVM to TFLu SHA

### PR DESCRIPTION
**Pull request by ashutosh-arm**
Aligning CMSIS-NN to TFLu's CMSIS-NN SHA. Since the
docker update will take some time, these code changes
contain support for old and new SHA both. After the
docker image update, we can remove the support with
older CMSIS-NN. This commit updates CMSIS-NN and
micronpu demos.

### CODE REVIEW

**Comment by ashutosh-arm** [comment]
cc: @grant-arm @Mousius @manupa-arm for code review.

File: apps/microtvm/cmsisnn/Makefile [hunk 1]
```diff
@@ -81,13 +81,27 @@ ${BUILD_DIR}/libcmsis_startup.a: $(CMSIS_STARTUP_SRCS)
 	$(QUIET)$(AR) -cr $(abspath $(BUILD_DIR)/libcmsis_startup.a) $(abspath $(BUILD_DIR))/libcmsis_startup/*.o
 	$(QUIET)$(RANLIB) $(abspath $(BUILD_DIR)/libcmsis_startup.a)
 
+CMSIS_SHA_FILE=${CMSIS_PATH}/977abe9849781a2e788b02282986480ff4e25ea6.sha
+ifneq ("$(wildcard $(CMSIS_SHA_FILE))","")
+${BUILD_DIR}/cmsis_nn/Source/libcmsis-nn.a:
+	$(QUIET)mkdir -p $(@D)
+	$(QUIET)cd $(CMSIS_PATH)/CMSIS/NN && $(CMAKE) -B $(abspath $(BUILD_DIR)/cmsis_nn) $(CMSIS_NN_CMAKE_FLAGS)
+	$(QUIET)cd $(abspath $(BUILD_DIR)/cmsis_nn) && $(MAKE) all
+else
```
Comment (grant-arm) [inline apps/microtvm/cmsisnn/Makefile:90]: For my own understanding, is this bit only necessary to get the PR through CI, before CMSIS-NN has been upgraded in the Docker image?

**Review by grant-arm — COMMENTED** [review]
Thanks for this PR @ashutosh-arm . It will be great to get the latest updates to CMSIS-NN included in TVM.

File: apps/microtvm/cmsisnn/Makefile (hunk 1 above)
Comment (ashutosh-arm) [reply apps/microtvm/cmsisnn/Makefile:90]: oh yes, for now we need to support with both docker images. We can get rid of this change once the new one available.

**Comment by grant-arm** [comment]
LGTM

File: docker/install/ubuntu_install_cmsis.sh [hunk 2]
```diff
@@ -35,15 +35,16 @@ fi
...
 
 # Download and extract CMSIS
-cd "${HOME}"
-wget --quiet "https://github.com/ARM-software/CMSIS_5/archive/${CMSIS_VER}.tar.gz"
-tar -xf "${CMSIS_VER}.tar.gz" -C "${INSTALLATION_PATH}" --strip-components=1
+CMSIS_SHA="977abe9849781a2e788b02282986480ff4e25ea6"
+CMSIS_URL="http://github.com/ARM-software/CMSIS_5/archive/${CMSIS_SHA}.tar.gz"
+DOWNLOAD_PATH="/tmp/${CMSIS_SHA}.tar.gz"
+
+wget ${CMSIS_URL} -O "${DOWNLOAD_PATH}"
```
Comment (Mousius) [inline docker/install/ubuntu_install_cmsis.sh:46]: Can we check the file SHA to ensure it's the file we expected?

Similar to:
https://github.com/apache/tvm/blob/70cc2abf91b5718b013fb4042dfe441d1918fcad/docker/install/ubuntu_install_boost.sh#L28-L29

**Review by Mousius — CHANGES_REQUESTED** [review]

File: docker/install/ubuntu_install_cmsis.sh (hunk 2 above)
Comment (ashutosh-arm) [reply docker/install/ubuntu_install_cmsis.sh:47]: Thanks @Mousius for pointing that out and helping with it.

**Comment by ashutosh-arm** [comment]
@Mousius PTAL again. Thanks!

**Comment by Mousius** [comment]
Thanks @ashutosh-arm, this is merged 😸 can you raise a Docker image issue for ci_cpu and ci_qemu?

**Pull request merged**